File: solucao-java/src/main/java/org/georgeprado/exercises/ex002.java
package org.georgeprado.exercises;

import java.util.Scanner;
import java.util.Arrays;

public class ex002 {
    public static void showAnswer() {
        Scanner scanner = new Scanner(System.in);

        System.out.print("Digite um número para verificar se pertence à sequência de Fibonacci: ");
        int numEntry = scanner.nextInt();

        int[] fibonacci = generateFibonacci(numEntry);

        System.out.printf("%d %s na sequência de Fibonacci.%n",
                numEntry, (isInFibonacci(numEntry, fibonacci) ? "está" : "NÃO está"));

        scanner.close();
    }

    public static int[] generateFibonacci(int numEntry) {
        int[] fibonacci = { 0, 1 };

        while (fibonacci[fibonacci.length - 1] < numEntry) {
            int nextTerm = fibonacci[fibonacci.length - 1] + fibonacci[fibonacci.length - 2];
            fibonacci = Arrays.copyOf(fibonacci, fibonacci.length + 1);
            fibonacci[fibonacci.length - 1] = nextTerm;
        }

        return fibonacci;
    }

    public static boolean isInFibonacci(int num, int[] fibonacci) {
        for (int fibo : fibonacci) {
            if (fibo == num) {
                return true;
            }
        }
        return false;
    }

    public static void main(String[] args) {
        showAnswer();
    }
}

File: solucao-csharp/Exercises/Ex002.cs
namespace desafio_target.Exercises
{
    internal class ExerciseTwo
    {
        public static void ShowAnswer()
        {
            Console.Write("Digite um número para verificar se pertence à sequência de Fibonacci: ");
            int numEntry = Convert.ToInt32(Console.ReadLine());

            int[] fibonacci = GenerateFibonacci(numEntry);

            Console.WriteLine($"{numEntry} {(IsInFibonacci(numEntry, fibonacci) ? "está" : "NÃO está")} na sequência de Fibonacci.");
        }

        public static int[] GenerateFibonacci(int numEntry)
        {
            int[] fibonacci = new int[2] { 0, 1 };

            while (fibonacci[^1] < numEntry)
            {
                int nextTerm = fibonacci[^1] + fibonacci[^2];
                Array.Resize(ref fibonacci, fibonacci.Length + 1);
                fibonacci[^1] = nextTerm;
            }

            return fibonacci;
        }

        public static bool IsInFibonacci(int num, int[] fibonacci)
        {
            foreach (int fibo in fibonacci)
            {
                if (fibo == num) return true;
            }

            return false;
        }
    }
}
